src/main/scala/tech/cryptonomic/conseil/tezos/TezosDatabaseOperations.scala

```diff
@@ -755,19 +757,68 @@ object TezosDatabaseOperations extends LazyLogging {
   def writeProcessedEventsLevels(eventType: String, levels: List[BigDecimal]): DBIO[Option[Int]] =
     Tables.ProcessedChainEvents ++= levels.map(Tables.ProcessedChainEventsRow(_, eventType))
 
-  /** Inserts registered tokens to the table if empty
-    *
+  /** Inserts to the table if table is empty
+    * @param table slick TableQuery[_] to which we want to insert
     * @param rows the type of event to record
     * @return the number of entries saved added to the
     */
-  def writeRegisteredTokensRowsIfEmpty(
-      rows: List[Tables.RegisteredTokensRow]
+  def insertIfEmptyTable[A <: AbstractTable[_]](
+      table: TableQuery[A],
+      rows: List[A#TableElementType]
   )(implicit ec: ExecutionContext): DBIO[Option[Int]] =
-    Tables.RegisteredTokens.exists.result.flatMap {
+    table.exists.result.flatMap {
       case true => DBIO.successful(Some(0))
-      case false => Tables.RegisteredTokens ++= rows
+      case false => table ++= rows
     }
 
+  import shapeless._
+  import shapeless.ops.hlist._
+
+  /** Trims if passed a String value, otherwise returns the value unchanged */
+  object Trimmer extends Poly1 {
+    implicit val stringTrim = at[String] { _.trim }
+    implicit def noop[T] = at[T] { identity }
+  }
+
+  /** Uses a Generic to transform the instance into an HList, maps over it and convert it back into the case class */
+  private def trimCaseClass[C, H <: HList](
+      c: C
+  )(implicit g: Generic.Aux[C, H], m: Mapper.Aux[Trimmer.type, H, H]): C = {
+    val hlist = g.to(c)
+    val trimmed = hlist.map(Trimmer)
+    g.from(trimmed)
+  }
+
+  import kantan.csv._
+  import kantan.csv.ops._
+
+  /** Reads and inserts CSV file to the database for the */
+  def initTableFromCsv[A <: AbstractTable[_], H <: HList](table: TableQuery[A], network: String, separator: Char = ',')(
+      implicit hd: HeaderDecoder[A#TableElementType],
+      g: Generic.Aux[A#TableElementType, H],
+      m: Mapper.Aux[Trimmer.type, H, H],
+      ec: ExecutionContext
+  ): Future[List[A#TableElementType]] = {
+
+    val rawData: java.net.URL = getClass.getResource(s"/${table.baseTableRow.tableName}/" + network + ".csv")
```

Comment:
Shouldn't this be passed in as a function parameter?

Reply:
I thought I'll use convention which Vishakh suggested in your ticket - he asked you to rename directory to name of the table(not exactly with this words, but this is what happened), so if we keep it this way we avoid duplicating information passed to this method

Reply:
Sorry for insisting, it's just that I already refactored that part in my PR, so we should decide how to proceed. Either mine or your PR will eventually need changes 😄

Reply:
Ok, I'd say leave it like that and we'll see what happens when I need to merge the other PR